### PR TITLE
Deployer contract migration

### PR DESCRIPTION
**Pull request by i-stam**
- migrateWallet()

- event name change

- public variables for generating getters

### CODE REVIEW

File: contracts/wallet-deployer.sol [hunk 1]
```diff
@@ -21,46 +21,63 @@ pragma solidity ^0.4.25;
...
+    }
+
+    function migrateWallet(address owner, uint _spendLimit, address[] _whitelistedAddresses) external onlyController {
+        if (cached.length < 1) {
+            cacheWallet();
+    	}
+        address walletAddress = cached[cached.length-1];
+        cached.length--;
+        deployed[owner] = walletAddress;
+        IWallet(walletAddress).initializeSpendLimit(_spendLimit);
```
Comment (draganm) [inline contracts/wallet-deployer.sol]: why not using Wallet(walletAddress)?

Comment (i-stam) [reply contracts/wallet-deployer.sol]: Fixed! we didn't really need the interface...

**Comment by i-stam** [comment]
Not sure that this change ensures user privacy, i think @draganm told me that we need this even during the migration phase....

**Comment by mischat** [comment]
@i-stam this has nothing to do with user privacy and go YOU ARE ON HOLIDAY TODAY ! speak tomorrow.

**Comment by mischat** [comment]
@draganm can you have a look at this please ?

File: contracts/wallet-deployer.sol [hunk 2]
```diff
@@ -21,46 +21,82 @@ pragma solidity ^0.4.25;
...
+/// @title Wallet Deployer
+/// @notice Factory method to create deploy a new wallet
+/// @notice Allows for pre-fabrication / caching of wallets
 contract WalletDeployer is Controllable {
 
-	event WalletCached(address wallet);
-	event WalletDeployed(address wallet, address owner);
+    event CachedWallet(address wallet);
+    event DeployedWallet(address wallet, address owner);
+    event MigratedWallet(address wallet, address owner);
```
Comment (draganm) [inline contracts/wallet-deployer.sol]: we should also add address of the previous wallet

File: test/wallet_deployer/migrate_wallet_test.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+				Expect(sl.String()).To(Equal(EthToWei(2).String()))
+			})
+
+			It("should update the initialization flag", func() {
+				initialized, err := Wallet.InitializedSpendLimit(nil)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(initialized).To(BeTrue())
+			})
+
+			It("should emit a topUp Gas set event", func() {
```
Comment (draganm) [inline test/wallet_deployer/migrate_wallet_test.go]: I'm pretty sure this text is wrong ...

File: test/wallet_deployer/migrate_wallet_test.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+				isWhitelisted, err := Wallet.IsWhitelisted(nil, common.HexToAddress("0x1"))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(isWhitelisted).To(BeTrue())
+
+				isWhitelisted, err = Wallet.IsWhitelisted(nil, common.HexToAddress("0x2"))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(isWhitelisted).To(BeTrue())
+			})
+
+			It("should emit WhitelistAddition event", func() {
```
Comment (draganm) [inline test/wallet_deployer/migrate_wallet_test.go]: event name is wrong in It()

File: test/wallet_deployer/migrate_wallet_test.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+		When("a controller migrates a Wallet", func() {
+			BeforeEach(func() {
+				tx, err := WalletDeployer.MigrateWallet(Controller.TransactOpts(), Owner.Address(), EthToWei(1), FinneyToWei(2), true, []common.Address{common.HexToAddress("0x1"), common.HexToAddress("0x2")})
+				Expect(err).ToNot(HaveOccurred())
+				Backend.Commit()
+				Expect(isSuccessful(tx)).To(BeTrue())
+			})
+
+			It("should emit MigratedContract event", func() {
```
Comment (draganm) [inline test/wallet_deployer/migrate_wallet_test.go]: MigratedWallet

**Pull request merged**